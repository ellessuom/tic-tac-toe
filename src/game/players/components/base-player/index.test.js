import React from 'react';
import { mount } from 'enzyme';
import BasePlayer from './index';

const fakeUseData = [
  {
    fake1: {
      wins: 3 // Random number, different from the default which is 0
    }
  }
];

const fakeLabels = {
  fake1: 'Player Fake'
};

jest.mock('./labels', () => fakeLabels);

jest.mock('../../../context', () => {
  return {
    useData: jest.fn(() => fakeUseData)
  };
});

describe('<BasePlayer />', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(
      <BasePlayer playerId="fake1" />
    );
  });

  it('should render <BasePlayer /> structure correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('<BasePlayer />\s children', () => {
    it('should have the expected title', () => {
      expect(wrapper.find('.title').text()).toBe(fakeLabels.fake1);
    });

    it('should have the expected value for wins', () => {
      expect(wrapper.find('.paragraph').text()).toBe(`Wins: ${fakeUseData[0].fake1.wins}`);
    });
  })
});
