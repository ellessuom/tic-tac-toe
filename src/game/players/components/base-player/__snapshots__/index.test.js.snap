// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasePlayer /> should render <BasePlayer /> structure correctly 1`] = `
<BasePlayer
  playerId="fake1"
>
  <div
    className="base-player-display fake1"
  >
    <Title>
      <div
        className="title"
      >
        Player Fake
      </div>
    </Title>
    <Paragraph>
      <div
        className="paragraph"
      >
        Wins: 
        3
      </div>
    </Paragraph>
  </div>
</BasePlayer>
`;
